internal/subshell/subshell.go

```diff
@@ -140,6 +142,45 @@ func New(cfg sscommon.Configurable) SubShell {
 	return subs
 }
 
+func availableShells() []SubShell {
+	var shells []SubShell
+	for _, shell := range supportedShells {
+		rcFile, err := shell.RcFile()
+		if err != nil {
+			logging.Error("Could not determine rc file for shell %s: %v", shell.Shell(), err)
+			continue
+		}
+
+		if !fileutils.FileExists(rcFile) {
+			continue
+		}
+
+		shells = append(shells, shell)
+	}
+
+	return shells
+}
+
+func ConfigureAvailableShells(cfg sscommon.Configurable, env map[string]string, identifier sscommon.RcIdentification, isAdmin bool) error {
+	activeShell := New(cfg)
+	for _, s := range availableShells() {
+		isActive := false
+		if s.Shell() == activeShell.Shell() {
+			isActive = true
+		}
+
+		err := s.WriteUserEnv(cfg, env, identifier, isAdmin, isActive)
```

Comment:
I think it would be cleaner if we introduced something like `subshell.IsAvailable()`.

So you'd end up with:

```
if s.IsAvailable() {
  s.WriteUserEnv(...)
}
```